lib/controllers/state_provider.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import '/controllers/app_bar_state.dart';
+import '/controllers/task_state.dart';
+import '/controllers/title_state.dart';
+
+class StateProvider extends StatefulWidget {
+  final Widget child;
+
+  const StateProvider({super.key, required this.child});
+
+  static StateProviderData of(BuildContext context) {
+    var data = context.dependOnInheritedWidgetOfExactType<_InheritedStateProvider>();
+    assert(data != null, 'Леее куда преш не видиш NotifierController в дереве виджетов нет');
+    return data!.data;
+  }
+
+  @override
+  State<StateProvider> createState() => StateProviderData();
+}
+
+class StateProviderData extends State<StateProvider> {
+  late final TaskState taskState;
+  late final AppBarState appBarState;
+  late final TitleState titleState;
+
+  @override
+  void initState() {
+    appBarState = AppBarState();
+    taskState = TaskState(appBarState: appBarState);
+    titleState = TitleState();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    appBarState.dispose();
+    taskState.dispose();
+
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _InheritedStateProvider(data: this, child: widget.child);
+  }
+}
+
+class _InheritedStateProvider extends InheritedWidget {
+  const _InheritedStateProvider({required this.data, required super.child});
+
+  final StateProviderData data;
+
+  @override
+  bool updateShouldNotify(_InheritedStateProvider oldWidget) {
+    return data.titleState != oldWidget.data.titleState ||
+        data.appBarState != oldWidget.data.appBarState ||
+        data.taskState != oldWidget.data.taskState;
```

Comment:
Ваш `ChangeNotifier` не работает так, как вы этого ожидаете, потому что вы в провайдере на него не подписываетесь. Отсюда и все костыли с сетстейтами и `AnimationBuilder`. Посмотрите [InheritedNotifier](https://api.flutter.dev/flutter/widgets/InheritedNotifier-class.html). А вообще, проще использовать [Provider](https://pub.dev/packages/provider)
```suggestion
        data.taskState != oldWidget.data.taskState;
```

Reply:
Провайдер я видел и не захотел его использовать для лучшего понимания фреймворка и внутренних процессов в нем (все равно насколько я понимаю дальше мы будем на BLOC переписывать)
Касательно InheritedNotifier спасибо за наводку
Вся информация о управлении состоянием которую я нашел использовала другие библиотеки либо была древней (настолько древней что видимо этих виджетов еще просто не было)